runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/vm/IfCommand.java

```diff
@@ -54,11 +54,29 @@ protected void execute(Runtime runtime, State state, ThreadId threadId) {
         EvalContext evalContext = ecf.global(ctx);
 
         ExpressionEvaluator ee = runtime.getService(ExpressionEvaluator.class);
-        Boolean ifResult = ee.eval(evalContext, expr, Boolean.class);
-        if (ifResult != null && ifResult) {
+        Object ifResult = ee.eval(evalContext, expr, Object.class);
+        if (isTrue(ifResult)) {
             frame.push(thenCommand);
         } else if (elseCommand != null) {
             frame.push(elseCommand);
         }
     }
+
+    private static boolean isTrue(Object value) {
+        if (value == null) {
+            return false;
+        }
+
+        if (value instanceof Boolean b) {
+            return b;
+        } else if (value instanceof String s) {
+            if ("true".equalsIgnoreCase(s)) {
```

Comment:
Do we really need to coerce string values "true/false" to boolean? I don't think we do that anywhere else?